packages/compiler/test/js-sourcemaps/complex-frontmatter.ts

```diff
@@ -0,0 +1,46 @@
+import { test } from 'uvu';
+import * as assert from 'uvu/assert';
+import { testJSSourcemap } from '../utils';
+
+const input = `---
+// TODO: Due to this issue: https://github.com/withastro/astro/issues/1438, this route can't be in the same folder
```

Comment:
an aside, for <https://defined.net> we get around this by generating both paths via the same `getStaticPaths` function && adding an `isPost: true` identifier to the post entries, then in the astro body:

```astro
{props.content.isPost ?
  <BlogPost content={content} />
  : <BlogIndex content={content} />}
```

Reply:
Ah! I wonder if we can fix that...

This was just a copy + paste from an existing test fixture. Can't remember where it came from. https://github.com/withastro/compiler/blob/main/packages/compiler/test/tsx-sourcemaps/complex-generics.ts#L6-L8

Reply:
```astro
---
import BlogIndexLayout from "@layouts/BlogIndex.astro";
import BlogPostLayout from "@layouts/BlogPost.astro";
import { assert } from "assert-ts";
import type { Page, PaginateFunction } from "astro";
import type { AstroComponentFactory } from "astro/dist/runtime/server";
import type { MarkdownFrontmatter } from "@data/posts/MarkdownFrontmatter";
import type { MarkdownOrMDXInstance } from "./_getAllPublishablePosts";
import BlogMeta from "./meta.json";

export async function getStaticPaths({ paginate }: { paginate: PaginateFunction }) {
  const { getAllPublishablePosts } = await import("./_getAllPublishablePosts");
  const allPages = getAllPublishablePosts();

  // generate paths for the blog posts
  const postPaths = allPages.map(({ Content, frontmatter }: MarkdownOrMDXInstance) => {
    return {
      params: {
        page: frontmatter.slug,
      },
      props: {
        page: {
          ...frontmatter,
          Content,
          isPost: true,
        },
      },
    };
  });

  // generate paths for the blog index pages
  // @ts-expect-error the return type of `paginate` is too generic here, so we use `unknown` to shut up ts
  const paginationPaths = paginate(allPages, { pageSize: 10 }) as Page<MarkdownOrMDXInstance>[];

  return [...paginationPaths, ...postPaths];
}

type Post = MarkdownFrontmatter & { Content: AstroComponentFactory; isPost: true };
type PaginationPage = Page<MarkdownOrMDXInstance> & { isPost: undefined };
type Props = {
  page: Post | PaginationPage;
};
const { page } = Astro.props as Props;
---

{
  page.isPost ? (
    <BlogPostLayout content={page}>
      <page.Content />
    </BlogPostLayout>
  ) : (
    <BlogIndexLayout
      meta={{ title: assert(BlogMeta.title), description: assert(BlogMeta.description) }}
      page={page}
    />
  )
}
```

:D